### ISSUE TITLE
xmlrpc throws "OverflowError: int exceeds XML-RPC limits" on 64bit integers

### ISSUE THREAD

**Issue by ghost** [python/cpython#102927]
The xmlrpc library does not support sending 64bit integers. When called with an integer larger than `MAXINT = 2147483647` or smaller than `MININT = -2147483648`, an `OverflowError: int exceeds XML-RPC limits` exception is thrown. 

Alternatively, one could make use of the <i8> type of XML-RPC. 

### Linked PRs
* gh-102930

**Comment by ronaldoussoren** [comment]
Both [wikipedia](http://xmlrpc.com/spec.md) and the [xmlrpc spec[(http://xmlrpc.com/spec.md) mention that integers are 4-byte integers and don't specify a wider integer type (like the ``<i8>`` you use in the PR).  Is there a newer spec that specifies larger integers?

**Comment by RethinkMatze** [comment]
Actually, I could not find an official specification that includes i8 etc., but it seems like [xmlrpc-c](https://xmlrpc-c.sourceforge.net/) and [java](https://ws.apache.org/xmlrpc/types.html) (?) and, last not least the [python implementation itself]( https://docs.python.org/3/library/xmlrpc.client.html) knows about that datatype. Also, the python client will handle the i8 tag on the receiving end.

![image](https://user-images.githubusercontent.com/117762426/227947113-6e62186c-392c-4d25-adf2-b4f5f558aa74.png)

Unfortunately, we need this functionality, as some ROS1 tools depend on it. Funnily enough, this "OverflowException" seem to have got added at some point in time, i.e. older versions of python do not throw.

**Comment by ctheune** [comment]
We just stumbled over the same thing. In Python 3.6 the support for i8/biginteger etc. was introduced, but only on the parsing side, not the dumping side. 

We currently start sending long integers as strings, which is insane.

I could imagine that this becomes a knob like the support for `allow_none` to add `allow_long` and then dump them with the `<i8>` if they overflow which seems to be the most widely supported case AFAICT. Alternatively we could make that configurable allowing to specific which tag to use so that `allow_long="i8"` would specificy that.

Not sure what the default should be. Currently it crashes on long ints, so I could imagine we allow `i8` as the default and would be more symmetric with our input/output.

**Comment by thisch** [comment]
Unfortunately the linked PR hasn't yet been reviewed by anyone from the automatically selected reviewers. 

Likewise, I've also seen in a different xmlrpc related issue/PR that there is a lack of interest from the core cpython devs - Not sure what we can do.

**Comment by ctheune** [comment]
One could always created a fork in a separate package as has been done in the past to drive/showcase innovation.

**Comment by serhiy-storchaka** [comment]
I agree that we need new options to enable these extension XML-RPC types, similar to `allow_none`.

Taking `allow_none` (instead of `allow_nil`) as a precedence, `allow_decimal` can be used for `<bigdecimal>`. But what should we choose for `<i8>` and `<biginteger>`?

**Comment by thisch** [comment]
> One could always created a fork in a separate package as has been done in the past to drive/showcase innovation.

Yes, this is a good idea. I could create a new xmlrpclib2 pkg that initially is just a copy of the version in the stdlib and then we improve it from there.

**Comment by encukou** [comment]
Yes, the `xmlrpclib2` route is probably better for everyone involved. You won't have to wait for the next release, and CPython won't have to merge unproven code.

As for the flag, you could choose `allow_biginteger` (which would either allow `<i8>` and `<biginteger>`, or it would *only* send `<biginteger>` -- the other side typically needs to do range validation anyway; special-casing a number of bytes seems outdated.)
